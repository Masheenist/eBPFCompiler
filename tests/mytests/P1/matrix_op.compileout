
Test File [/home/masheenist/source/lab4-lab4-hanscom-llosa/tests/mytests/matrix_op.py]
- - - - - - - -
# matrix operations
matrix1 = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
matrix2 = [[1, 4, 7], [2, 5, 8], [3, 6, 9]]

# Matrix Addition
# output: [[2, 6, 10], [6, 10, 14], [10, 14, 18]]
matrix1[0][0] = matrix1[0][0] + matrix2[0][0]
matrix1[0][1] = matrix1[0][1] + matrix2[0][1]
matrix1[0][2] = matrix1[0][2] + matrix2[0][2]
matrix1[1][0] = matrix1[1][0] + matrix2[1][0]
matrix1[1][1] = matrix1[1][1] + matrix2[1][1]
matrix1[1][2] = matrix1[1][2] + matrix2[1][2]
matrix1[2][0] = matrix1[2][0] + matrix2[2][0]
matrix1[2][1] = matrix1[2][1] + matrix2[2][1]
matrix1[2][2] = matrix1[2][2] + matrix2[2][2]

print matrix1

# Matrix Trace
# output: 15
print matrix2[0][0] + matrix2[1][1] + matrix2[2][2]


# Matrix Transpose of matrix2
# output: True

matrix2_t2 = [[matrix2[2][0], matrix2[1][0], matrix2[0][0]],
                [matrix2[2][1], matrix2[1][1], matrix2[0][1]],
                [matrix2[2][2], matrix2[1][2], matrix2[0][2]]]


tmp0 = matrix2[0][0]
matrix2[0][0] = matrix2[2][0]
tmp2 = matrix2[0][1]
matrix2[0][1] = matrix2[1][0]
tmp3 = matrix2[0][2]
matrix2[0][2] = tmp0
tmp4 = matrix2[1][0]
matrix2[1][0] = matrix2[2][1]
tmp5 = matrix2[1][2]
matrix2[1][2] = tmp2
tmp6 = matrix2[2][0]
tmp7 = matrix2[2][2]
matrix2[2][0] = tmp7
matrix2[2][1] = tmp5
matrix2[2][2] = tmp3

print matrix2 == matrix2_t2
- - - - - - - -
explicate Module(None, Stmt([Assign([AssName('matrix1', 'OP_ASSIGN')], List([List([InjectFrom(Const(0), Const(1)), InjectFrom(Const(0), Const(2)), InjectFrom(Const(0), Const(3))]), List([InjectFrom(Const(0), Const(4)), InjectFrom(Const(0), Const(5)), InjectFrom(Const(0), Const(6))]), List([InjectFrom(Const(0), Const(7)), InjectFrom(Const(0), Const(8)), InjectFrom(Const(0), Const(9))])])), Assign([AssName('matrix2', 'OP_ASSIGN')], List([List([InjectFrom(Const(0), Const(1)), InjectFrom(Const(0), Const(4)), InjectFrom(Const(0), Const(7))]), List([InjectFrom(Const(0), Const(2)), InjectFrom(Const(0), Const(5)), InjectFrom(Const(0), Const(8))]), List([InjectFrom(Const(0), Const(3)), InjectFrom(Const(0), Const(6)), InjectFrom(Const(0), Const(9))])])), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(0)), Let(Name('0'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(0))], None, None), Let(Name('1'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(0))], None, None), IfExp(Let(Name('5'), Compare(GetTag(Name('0')), [('==', Const(3))]), IfExp(Name('5'), Compare(GetTag(Name('1')), [('==', Const(3))]), Name('5'))), InjectFrom(Const(3), CallFunc('add', [ProjectTo(Const(3), Name('0')), ProjectTo(Const(3), Name('1'))], None, None)), IfExp(Let(Name('4'), Let(Name('2'), Compare(GetTag(Name('0')), [('==', Const(0))]), IfExp(Name('2'), Compare(GetTag(Name('1')), [('==', Const(0))]), Name('2'))), IfExp(Not(Name('4')), Let(Name('3'), Compare(GetTag(Name('0')), [('==', Const(1))]), IfExp(Name('3'), Compare(GetTag(Name('1')), [('==', Const(1))]), Name('3'))), Name('4'))), InjectFrom(Const(0), Add((ProjectTo(Const(0), Name('0')), ProjectTo(Const(0), Name('1'))))), CallFunc('abort', [], None, None)))))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(1)), Let(Name('6'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(1))], None, None), Let(Name('7'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(1))], None, None), IfExp(Let(Name('11'), Compare(GetTag(Name('6')), [('==', Const(3))]), IfExp(Name('11'), Compare(GetTag(Name('7')), [('==', Const(3))]), Name('11'))), InjectFrom(Const(3), CallFunc('add', [ProjectTo(Const(3), Name('6')), ProjectTo(Const(3), Name('7'))], None, None)), IfExp(Let(Name('10'), Let(Name('8'), Compare(GetTag(Name('6')), [('==', Const(0))]), IfExp(Name('8'), Compare(GetTag(Name('7')), [('==', Const(0))]), Name('8'))), IfExp(Not(Name('10')), Let(Name('9'), Compare(GetTag(Name('6')), [('==', Const(1))]), IfExp(Name('9'), Compare(GetTag(Name('7')), [('==', Const(1))]), Name('9'))), Name('10'))), InjectFrom(Const(0), Add((ProjectTo(Const(0), Name('6')), ProjectTo(Const(0), Name('7'))))), CallFunc('abort', [], None, None)))))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(2)), Let(Name('12'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(2))], None, None), Let(Name('13'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(2))], None, None), IfExp(Let(Name('17'), Compare(GetTag(Name('12')), [('==', Const(3))]), IfExp(Name('17'), Compare(GetTag(Name('13')), [('==', Const(3))]), Name('17'))), InjectFrom(Const(3), CallFunc('add', [ProjectTo(Const(3), Name('12')), ProjectTo(Const(3), Name('13'))], None, None)), IfExp(Let(Name('16'), Let(Name('14'), Compare(GetTag(Name('12')), [('==', Const(0))]), IfExp(Name('14'), Compare(GetTag(Name('13')), [('==', Const(0))]), Name('14'))), IfExp(Not(Name('16')), Let(Name('15'), Compare(GetTag(Name('12')), [('==', Const(1))]), IfExp(Name('15'), Compare(GetTag(Name('13')), [('==', Const(1))]), Name('15'))), Name('16'))), InjectFrom(Const(0), Add((ProjectTo(Const(0), Name('12')), ProjectTo(Const(0), Name('13'))))), CallFunc('abort', [], None, None)))))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(0)), Let(Name('18'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(0))], None, None), Let(Name('19'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(0))], None, None), IfExp(Let(Name('23'), Compare(GetTag(Name('18')), [('==', Const(3))]), IfExp(Name('23'), Compare(GetTag(Name('19')), [('==', Const(3))]), Name('23'))), InjectFrom(Const(3), CallFunc('add', [ProjectTo(Const(3), Name('18')), ProjectTo(Const(3), Name('19'))], None, None)), IfExp(Let(Name('22'), Let(Name('20'), Compare(GetTag(Name('18')), [('==', Const(0))]), IfExp(Name('20'), Compare(GetTag(Name('19')), [('==', Const(0))]), Name('20'))), IfExp(Not(Name('22')), Let(Name('21'), Compare(GetTag(Name('18')), [('==', Const(1))]), IfExp(Name('21'), Compare(GetTag(Name('19')), [('==', Const(1))]), Name('21'))), Name('22'))), InjectFrom(Const(0), Add((ProjectTo(Const(0), Name('18')), ProjectTo(Const(0), Name('19'))))), CallFunc('abort', [], None, None)))))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(1)), Let(Name('24'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(1))], None, None), Let(Name('25'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(1))], None, None), IfExp(Let(Name('29'), Compare(GetTag(Name('24')), [('==', Const(3))]), IfExp(Name('29'), Compare(GetTag(Name('25')), [('==', Const(3))]), Name('29'))), InjectFrom(Const(3), CallFunc('add', [ProjectTo(Const(3), Name('24')), ProjectTo(Const(3), Name('25'))], None, None)), IfExp(Let(Name('28'), Let(Name('26'), Compare(GetTag(Name('24')), [('==', Const(0))]), IfExp(Name('26'), Compare(GetTag(Name('25')), [('==', Const(0))]), Name('26'))), IfExp(Not(Name('28')), Let(Name('27'), Compare(GetTag(Name('24')), [('==', Const(1))]), IfExp(Name('27'), Compare(GetTag(Name('25')), [('==', Const(1))]), Name('27'))), Name('28'))), InjectFrom(Const(0), Add((ProjectTo(Const(0), Name('24')), ProjectTo(Const(0), Name('25'))))), CallFunc('abort', [], None, None)))))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(2)), Let(Name('30'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(2))], None, None), Let(Name('31'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(2))], None, None), IfExp(Let(Name('35'), Compare(GetTag(Name('30')), [('==', Const(3))]), IfExp(Name('35'), Compare(GetTag(Name('31')), [('==', Const(3))]), Name('35'))), InjectFrom(Const(3), CallFunc('add', [ProjectTo(Const(3), Name('30')), ProjectTo(Const(3), Name('31'))], None, None)), IfExp(Let(Name('34'), Let(Name('32'), Compare(GetTag(Name('30')), [('==', Const(0))]), IfExp(Name('32'), Compare(GetTag(Name('31')), [('==', Const(0))]), Name('32'))), IfExp(Not(Name('34')), Let(Name('33'), Compare(GetTag(Name('30')), [('==', Const(1))]), IfExp(Name('33'), Compare(GetTag(Name('31')), [('==', Const(1))]), Name('33'))), Name('34'))), InjectFrom(Const(0), Add((ProjectTo(Const(0), Name('30')), ProjectTo(Const(0), Name('31'))))), CallFunc('abort', [], None, None)))))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(0)), Let(Name('36'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(0))], None, None), Let(Name('37'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(0))], None, None), IfExp(Let(Name('41'), Compare(GetTag(Name('36')), [('==', Const(3))]), IfExp(Name('41'), Compare(GetTag(Name('37')), [('==', Const(3))]), Name('41'))), InjectFrom(Const(3), CallFunc('add', [ProjectTo(Const(3), Name('36')), ProjectTo(Const(3), Name('37'))], None, None)), IfExp(Let(Name('40'), Let(Name('38'), Compare(GetTag(Name('36')), [('==', Const(0))]), IfExp(Name('38'), Compare(GetTag(Name('37')), [('==', Const(0))]), Name('38'))), IfExp(Not(Name('40')), Let(Name('39'), Compare(GetTag(Name('36')), [('==', Const(1))]), IfExp(Name('39'), Compare(GetTag(Name('37')), [('==', Const(1))]), Name('39'))), Name('40'))), InjectFrom(Const(0), Add((ProjectTo(Const(0), Name('36')), ProjectTo(Const(0), Name('37'))))), CallFunc('abort', [], None, None)))))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(1)), Let(Name('42'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(1))], None, None), Let(Name('43'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(1))], None, None), IfExp(Let(Name('47'), Compare(GetTag(Name('42')), [('==', Const(3))]), IfExp(Name('47'), Compare(GetTag(Name('43')), [('==', Const(3))]), Name('47'))), InjectFrom(Const(3), CallFunc('add', [ProjectTo(Const(3), Name('42')), ProjectTo(Const(3), Name('43'))], None, None)), IfExp(Let(Name('46'), Let(Name('44'), Compare(GetTag(Name('42')), [('==', Const(0))]), IfExp(Name('44'), Compare(GetTag(Name('43')), [('==', Const(0))]), Name('44'))), IfExp(Not(Name('46')), Let(Name('45'), Compare(GetTag(Name('42')), [('==', Const(1))]), IfExp(Name('45'), Compare(GetTag(Name('43')), [('==', Const(1))]), Name('45'))), Name('46'))), InjectFrom(Const(0), Add((ProjectTo(Const(0), Name('42')), ProjectTo(Const(0), Name('43'))))), CallFunc('abort', [], None, None)))))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(2)), Let(Name('48'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix1'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(2))], None, None), Let(Name('49'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(2))], None, None), IfExp(Let(Name('53'), Compare(GetTag(Name('48')), [('==', Const(3))]), IfExp(Name('53'), Compare(GetTag(Name('49')), [('==', Const(3))]), Name('53'))), InjectFrom(Const(3), CallFunc('add', [ProjectTo(Const(3), Name('48')), ProjectTo(Const(3), Name('49'))], None, None)), IfExp(Let(Name('52'), Let(Name('50'), Compare(GetTag(Name('48')), [('==', Const(0))]), IfExp(Name('50'), Compare(GetTag(Name('49')), [('==', Const(0))]), Name('50'))), IfExp(Not(Name('52')), Let(Name('51'), Compare(GetTag(Name('48')), [('==', Const(1))]), IfExp(Name('51'), Compare(GetTag(Name('49')), [('==', Const(1))]), Name('51'))), Name('52'))), InjectFrom(Const(0), Add((ProjectTo(Const(0), Name('48')), ProjectTo(Const(0), Name('49'))))), CallFunc('abort', [], None, None)))))], None, None)), Discard(CallFunc('print_any', [Name('matrix1')], None, None)), Discard(CallFunc('print_any', [Let(Name('54'), Let(Name('60'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(0))], None, None), Let(Name('61'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(1))], None, None), IfExp(Let(Name('65'), Compare(GetTag(Name('60')), [('==', Const(3))]), IfExp(Name('65'), Compare(GetTag(Name('61')), [('==', Const(3))]), Name('65'))), InjectFrom(Const(3), CallFunc('add', [ProjectTo(Const(3), Name('60')), ProjectTo(Const(3), Name('61'))], None, None)), IfExp(Let(Name('64'), Let(Name('62'), Compare(GetTag(Name('60')), [('==', Const(0))]), IfExp(Name('62'), Compare(GetTag(Name('61')), [('==', Const(0))]), Name('62'))), IfExp(Not(Name('64')), Let(Name('63'), Compare(GetTag(Name('60')), [('==', Const(1))]), IfExp(Name('63'), Compare(GetTag(Name('61')), [('==', Const(1))]), Name('63'))), Name('64'))), InjectFrom(Const(0), Add((ProjectTo(Const(0), Name('60')), ProjectTo(Const(0), Name('61'))))), CallFunc('abort', [], None, None))))), Let(Name('55'), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(2))], None, None), IfExp(Let(Name('59'), Compare(GetTag(Name('54')), [('==', Const(3))]), IfExp(Name('59'), Compare(GetTag(Name('55')), [('==', Const(3))]), Name('59'))), InjectFrom(Const(3), CallFunc('add', [ProjectTo(Const(3), Name('54')), ProjectTo(Const(3), Name('55'))], None, None)), IfExp(Let(Name('58'), Let(Name('56'), Compare(GetTag(Name('54')), [('==', Const(0))]), IfExp(Name('56'), Compare(GetTag(Name('55')), [('==', Const(0))]), Name('56'))), IfExp(Not(Name('58')), Let(Name('57'), Compare(GetTag(Name('54')), [('==', Const(1))]), IfExp(Name('57'), Compare(GetTag(Name('55')), [('==', Const(1))]), Name('57'))), Name('58'))), InjectFrom(Const(0), Add((ProjectTo(Const(0), Name('54')), ProjectTo(Const(0), Name('55'))))), CallFunc('abort', [], None, None)))))], None, None)), Assign([AssName('matrix2_t2', 'OP_ASSIGN')], List([List([CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(0))], None, None), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(0))], None, None), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(0))], None, None)]), List([CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(1))], None, None), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(1))], None, None), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(1))], None, None)]), List([CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(2))], None, None), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(2))], None, None), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(2))], None, None)])])), Assign([AssName('tmp0', 'OP_ASSIGN')], CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(0))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(0)), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(0))], None, None)], None, None)), Assign([AssName('tmp2', 'OP_ASSIGN')], CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(1))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(1)), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(0))], None, None)], None, None)), Assign([AssName('tmp3', 'OP_ASSIGN')], CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(2))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(0))], None, None), InjectFrom(Const(0), Const(2)), Name('tmp0')], None, None)), Assign([AssName('tmp4', 'OP_ASSIGN')], CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(0))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(0)), CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(1))], None, None)], None, None)), Assign([AssName('tmp5', 'OP_ASSIGN')], CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(2))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(1))], None, None), InjectFrom(Const(0), Const(2)), Name('tmp2')], None, None)), Assign([AssName('tmp6', 'OP_ASSIGN')], CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(0))], None, None)), Assign([AssName('tmp7', 'OP_ASSIGN')], CallFunc('get_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(2))], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(0)), Name('tmp7')], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(1)), Name('tmp5')], None, None)), Discard(CallFunc('set_subscript', [CallFunc('get_subscript', [Name('matrix2'), InjectFrom(Const(0), Const(2))], None, None), InjectFrom(Const(0), Const(2)), Name('tmp3')], None, None)), Discard(CallFunc('print_any', [Let(Name('66'), Name('matrix2'), Let(Name('67'), Name('matrix2_t2'), IfExp(Let(Name('71'), Compare(GetTag(Name('66')), [('==', Const(3))]), IfExp(Name('71'), Compare(GetTag(Name('67')), [('==', Const(3))]), Name('71'))), InjectFrom(Const(1), CallFunc('equal', [ProjectTo(Const(3), Name('66')), ProjectTo(Const(3), Name('67'))], None, None)), IfExp(Let(Name('70'), Let(Name('68'), Compare(GetTag(Name('66')), [('==', Const(0))]), IfExp(Name('68'), Compare(GetTag(Name('67')), [('==', Const(0))]), Name('68'))), IfExp(Not(Name('70')), Let(Name('69'), Compare(GetTag(Name('66')), [('==', Const(1))]), IfExp(Name('69'), Compare(GetTag(Name('67')), [('==', Const(1))]), Name('69'))), Name('70'))), InjectFrom(Const(1), Compare(ProjectTo(Const(0), Name('66')), [('==', ProjectTo(Const(0), Name('67')))])), CallFunc('abort', [], None, None)))))], None, None))]))

