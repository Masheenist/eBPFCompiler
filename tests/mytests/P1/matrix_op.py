# matrix operations
matrix1 = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
matrix2 = [[1, 4, 7], [2, 5, 8], [3, 6, 9]]

# Matrix Addition
# output: [[2, 6, 10], [6, 10, 14], [10, 14, 18]]
matrix1[0][0] = matrix1[0][0] + matrix2[0][0]
matrix1[0][1] = matrix1[0][1] + matrix2[0][1]
matrix1[0][2] = matrix1[0][2] + matrix2[0][2]
matrix1[1][0] = matrix1[1][0] + matrix2[1][0]
matrix1[1][1] = matrix1[1][1] + matrix2[1][1]
matrix1[1][2] = matrix1[1][2] + matrix2[1][2]
matrix1[2][0] = matrix1[2][0] + matrix2[2][0]
matrix1[2][1] = matrix1[2][1] + matrix2[2][1]
matrix1[2][2] = matrix1[2][2] + matrix2[2][2]

print matrix1

# Matrix Trace
# output: 15
print matrix2[0][0] + matrix2[1][1] + matrix2[2][2]


# Matrix Transpose of matrix2
# output: True

matrix2_t2 = [[matrix2[2][0], matrix2[1][0], matrix2[0][0]],
                [matrix2[2][1], matrix2[1][1], matrix2[0][1]],
                [matrix2[2][2], matrix2[1][2], matrix2[0][2]]]


tmp0 = matrix2[0][0]
matrix2[0][0] = matrix2[2][0]
tmp2 = matrix2[0][1]
matrix2[0][1] = matrix2[1][0]
tmp3 = matrix2[0][2]
matrix2[0][2] = tmp0
tmp4 = matrix2[1][0]
matrix2[1][0] = matrix2[2][1]
tmp5 = matrix2[1][2]
matrix2[1][2] = tmp2
tmp6 = matrix2[2][0]
tmp7 = matrix2[2][2]
matrix2[2][0] = tmp7
matrix2[2][1] = tmp5
matrix2[2][2] = tmp3

print matrix2 == matrix2_t2
