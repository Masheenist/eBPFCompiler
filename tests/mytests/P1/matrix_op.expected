[[2, 6, 10], [6, 10, 14], [10, 14, 18]]
15
True
